llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -928,6 +932,7 @@ void BlockState::classify() {
   if (LoopAware && IsLoop(TheBlock) &&
       llvm::all_of(TheBlock->successors(), CanFixLoopSchedule)) {
     Kind = BlockType::Loop;
+    FixPoint.Stage = SchedulingStage::GatheringRegions;
```

Comment:
nit: I have a feeling that it would be more consistent to always gather regions in the first sweep, also for regular blocks But I agree that would introduce iterating on non-loop blocks. Not now.